src/config.rs

```diff
@@ -555,8 +565,9 @@ impl Config {
       Choose { overrides, chooser } =>
         self.choose(justfile, &search, overrides, chooser.as_deref())?,
       Command { overrides, .. } => self.run(justfile, &search, overrides, &[])?,
-      Dump => Self::dump(justfile),
+      Dump => self.dump(&src)?,
```

Comment:
We're currently compiling twice, which would be nice to avoid.

Instead of calling `Compiler::compile`, let's do something like:

```rust
    let tokens = Lexer::lex(src).eprint(self.color)?;
    let ast = Parser::parse(&tokens).eprint(self.color)?;
    let justfile = Analyzer::analyze(ast.clone()).eprint(self.color)?;
```

Then, the AST can be passed into `dump` and `format`, so we don't have to recompile.